### PR TITLE
Android ci

### PR DESCRIPTION
**Pull request by BrzVlad**
Facilitates automated build and test running on jenkins

### CODE REVIEW

File: android/Makefile [hunk 1]
```diff
@@ -53,7 +53,7 @@ kill:
 	$(ADB) shell am force-stop $(PACKAGE)
 
 $(PROJECT_SETUP_FILES): build.xml AndroidManifest.xml
-	$(ANDROID) update project -p . -t "android-14"
+	$(ANDROID) update project -p . -t "android-21"
```
Comment (lewurm) [inline android/Makefile:56]: why is this needed? I think `xamarin-android` supports API 14 as a minimum, we might want that as well.

File: android/tests/Makefile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+PORT = 5600
+IMAGE_NAME = AndroidEmulatorMono
+SDK_VERSION = 21
+ANDROID_ABI = armeabi-v7a
+TARGET_ID = android-21
+
+PROJECT_SETUP_FILES = project.properties local.properties
+
+app: $(PROJECT_SETUP_FILES)
```
Comment (lewurm) [inline android/tests/Makefile]: kinda dirty, add `.PHONY: app test emulator clean` at least

File: install-dependencies.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+platform-tools_r23-linux.zip sdk/platform-tools
+build-tools_r25.0.1-linux.zip sdk/build-tools/25.0.1
+armeabi-v7a-21_r04.zip sdk/system-images/android-21/armeabi-v7a /sys-img/android"
+
+ANDROID_MAC_DEPS="android-ndk-r11c-darwin-x86_64.zip ndk
+tools_r24.4.1-macosx.zip sdk/tools
+android-21_r02.zip sdk/platforms/android-21
+platform-tools_r23-macosx.zip sdk/platform-tools
+build-tools_r25.0.1-macosx.zip sdk/build-tools/25.0.1
+armeabi-v7a-21_r04.zip sdk/system-images/android-21/armeabi-v7a /sys-img/android"
```
Comment (lewurm) [inline install-dependencies.sh:15]: adding checksums would be nice too.

**Comment by lewurm** [comment]
with this PR, what is jenkins required to execute?

File: android/Makefile (hunk 1 above)
Comment (BrzVlad) [reply android/Makefile:56]: @lewurm this was done to avoid having to download both sdk platforms 14 and 21. For whatever reason, in android-template.rb we require up to android-21 platform. I'm assuming there should be no relevant difference if using a newer sdk.

**Comment by BrzVlad** [comment]
@lewurm On jenkins : 

```
make
make -C android/tests emulator
make -C android/tests test
```

File: android/Makefile (hunk 1 above)
Comment (lewurm) [reply android/Makefile:56]: @jonpryor @kumpera what do you think? can we go with android-21?

**Comment by luhenry** [comment]
This repo has been moved to mono `sdks/`, and AFAICT most of the features have been ported their already. @BrzVlad  do you think there is any outstanding feature here that has not been ported yet in mono's `sdks/`?
